metastats-domain/src/main/groovy/life/qbic/metastats/filter/FilterExperimentData.groovy

```diff
@@ -128,20 +122,16 @@ class FilterExperimentData implements FilterExperimentDataInput {
      * Method to calculate the sequencing mode of samples based on the filenames
      * @param samples
      */
-    void createSequencingModeEntry(List<MetaStatsSample> samples) {
+    void createSequencingModeEntry(List<MetaStatsSample> samples){
         samples.each { sample ->
             String filename = sample.sampleProperties.get("Filename")
+            String sequencingMode = SequencingModeCalculator.calculateSequencingMode(filename)
```

Comment:
Probably better to throw an exception in the SequencingModeCalculator, in case the sequencing mode calculation did not work. This exception can then be thrown below and you avoid the ugly `""== variable` comparison.

Reply:
I can see your point. But I tried to avoid throwing an exception since otherwise the sheet will not be generated. I just inform the user about wrong filenames

Reply:
```suggestion
            String sequencingMode
            try {
               sequencingMode = SequencingModeCalculator.calculateSequencingMode(filename)
            } catch (IllegalArgumentException e) {
               sequencingMode = ""
               messageOutputPort.invokeOnError("The sequencing mode cannot be defined for $sample.sampleCode", e)
            }
```

This way you keep you original exception, its class as well as associated causes.